eth/api_backend.go

```diff
@@ -74,7 +74,7 @@ func (b *EthApiBackend) StateAndHeaderByNumber(blockNr rpc.BlockNumber) (ethapi.
 	// Pending state is only known by the miner
 	if blockNr == rpc.PendingBlockNumber {
 		block, state := b.eth.miner.Pending()
-		return EthApiState{state}, block.Header(), nil
+		return EthApiState{state.Copy()}, block.Header(), nil
```

Comment:
The copy can also be made in `miner.Pending()` itself. That would prevents similar errors when other parts of the code base try to (mis)use the pending state.